test/Microsoft.DotNet.Watcher.Tools.FunctionalTests/Scenario/ProjectToolScenario.cs

```diff
@@ -26,57 +32,34 @@ public ProjectToolScenario()
             CreateTestDirectory();
         }
 
-        public string TempFolder { get; } = Path.Combine(Path.GetDirectoryName(FindNugetConfig()), "testWorkDir", Guid.NewGuid().ToString());
+        public string TempFolder { get; } = Path.Combine(Path.GetDirectoryName(FindNugetConfig()), "testWorkDir", Guid.NewGuid().ToString("N"));
 
         public string WorkFolder { get; }
 
-        public void AddProject(string projectFolder)
+        public void AddTestProjectFolder(string projectName)
         {
-            var destinationFolder = Path.Combine(WorkFolder, Path.GetFileName(projectFolder));
-            Console.WriteLine($"Copying project {projectFolder} to {destinationFolder}");
+            var srcFolder = Path.Combine(TestProjectSourceRoot, projectName);
+            var destinationFolder = Path.Combine(WorkFolder, Path.GetFileName(projectName));
+            Console.WriteLine($"Copying project {srcFolder} to {destinationFolder}");
 
             Directory.CreateDirectory(destinationFolder);
 
-            foreach (var directory in Directory.GetDirectories(projectFolder, "*", SearchOption.AllDirectories))
+            foreach (var directory in Directory.GetDirectories(srcFolder, "*", SearchOption.AllDirectories))
             {
-                Directory.CreateDirectory(directory.Replace(projectFolder, destinationFolder));
+                Directory.CreateDirectory(directory.Replace(srcFolder, destinationFolder));
             }
 
-            foreach (var file in Directory.GetFiles(projectFolder, "*.*", SearchOption.AllDirectories))
+            foreach (var file in Directory.GetFiles(srcFolder, "*", SearchOption.AllDirectories))
             {
-                File.Copy(file, file.Replace(projectFolder, destinationFolder), true);
-            }
-        }
-
-        public void AddNugetFeed(string feedName, string feed)
-        {
-            var tempNugetConfigFile = Path.Combine(WorkFolder, NugetConfigFileName);
-
-            var nugetConfig = XDocument.Load(tempNugetConfigFile);
-            var packageSource = nugetConfig.Element("configuration").Element("packageSources");
-            packageSource.Add(new XElement("add", new XAttribute("key", feedName), new XAttribute("value", feed)));
-            using (var stream = File.OpenWrite(tempNugetConfigFile))
-            {
-                nugetConfig.Save(stream);
+                var dest = file.Replace(srcFolder, destinationFolder);
+                if ("P.json".Equals(Path.GetFileName(file), StringComparison.OrdinalIgnoreCase))
```

Comment:
Does it add too much to the build? I wonder it would not be better just to restore for these projects instead of introducing P.json which no one will be familiar with.